docker/Dockerfile-corsika-simtel

```diff
@@ -76,41 +77,67 @@ WORKDIR /workdir/sim_telarray/corsika-$CORSIKA_VERSION
 RUN if [ "$CORSIKA_VERSION" = "77550" ]; then \
         patch -b -p0 src/corsika.F < "../corsika-77550.patch"; \
     fi && \
-    if [ "$AVX_FLAG" = "no_opt" ]; then \
-        ../corsika_build_script "$HADRONIC_MODEL" generic; \
-    else \
+    if [ "$AVX_FLAG" != "no_opt" ]; then \
         ./corsika_opt_patching.sh && \
-        autoreconf && \
-        ../corsika_build_script "$HADRONIC_MODEL" generic cerenkopt vlibm; \
+        autoreconf; \
     fi
 
 # Build non-optimized / optimized CORSIKA
 WORKDIR /workdir/sim_telarray
-RUN make -C "corsika-$CORSIKA_VERSION" clean && rm -f ./*/*.a lib/*/*.a && \
-    AVX_EXTRA_FLAG=$([ "$AVX_FLAG" = "avx512f" ] && echo "-ffp-contract=off" || echo "") && \
-    AVX_EXTRA_FLAG="" && \
-    CFLAGS="" && CXXFLAGS="" && FFLAGS="" && \
-    if [ "$AVX_FLAG" = "no_opt" ]; then \
-        CFLAGS="-g  -std=c99 -O3"; \
-        CXXFLAGS="-g  -std=c99 -O3"; \
-        FFLAGS="-g -ffixed-line-length-132 -fno-automatic -frecord-marker=4 -std=legacy -O3"; \
-    else \
-        CFLAGS="-g -DCERENKOPT -DVLIBM -std=c99 -O3 ${AVX_FLAG:+-m$AVX_FLAG} ${AVX_EXTRA_FLAG} -DVECTOR_SIZE=8 -DVECTOR_LENGTH=8"; \
-        CXXFLAGS="-g -DCERENKOPT -DVLIBM -std=c99 -O3 ${AVX_FLAG:+-m$AVX_FLAG} ${AVX_EXTRA_FLAG} -DVECTOR_SIZE=8 -DVECTOR_LENGTH=8"; \
-        FFLAGS="-g -ffixed-line-length-132 -fno-automatic -frecord-marker=4 -std=legacy -DCERENKOPT -DVLIBM -std=c99 -O3 ${AVX_FLAG:+-m$AVX_FLAG} ${AVX_EXTRA_FLAG} -DVECTOR_SIZE=8 -DVECTOR_LENGTH=8"; \
-    fi && \
-    make -C "corsika-$CORSIKA_VERSION" \
-        CC="${GCC_PATH}/gcc -static" \
-        CXX="${GCC_PATH}/g++ -static" \
-        CPP="${GCC_PATH}/gcc -static -E" \
-        F77="${GCC_PATH}/gfortran -L/lib64/" \
-        CFLAGS="${CFLAGS}" \
-        CXXFLAGS="${CXXFLAGS}" \
-        FFLAGS="${FFLAGS}" \
-        install
-
-# Rename executable to 'corsika'
-RUN mv -f "corsika-${CORSIKA_VERSION}/run/corsika${CORSIKA_VERSION}Linux_"* "corsika-${CORSIKA_VERSION}/run/corsika"
+# hadolint ignore=DL3003
+RUN for variant in curved ""; do \
+        (cd ./corsika-$CORSIKA_VERSION && \
+        if [ "$AVX_FLAG" = "no_opt" ] || [ "$variant" = "curved" ]; then \
+            ../corsika_build_script "$HADRONIC_MODEL" generic $variant; \
+        else \
+            ../corsika_build_script "$HADRONIC_MODEL" generic cerenkopt vlibm $variant; \
+        fi) && \
+        make -C "corsika-$CORSIKA_VERSION" clean && rm -f ./*/*.a lib/*/*.a && \
+        AVX_EXTRA_FLAG=$([ "$AVX_FLAG" = "avx512f" ] && echo "-ffp-contract=off" || echo "") && \
+        AVX_EXTRA_FLAG="" && \
+        CFLAGS="" && CXXFLAGS="" && FFLAGS="" && \
+        if [ "$AVX_FLAG" = "no_opt" ] || [ "$variant" = "curved" ]; then \
+            CFLAGS="-g  -std=c99 -O3"; \
```

Comment:
CFLAGS/CXXFLAGS/FFLAGS have a lot in common, maybe you can define a base flag that is passed to them.